### PR TITLE
Make the userdir synced not rely on being able to search for yourself

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

File: tests/52user-directory/01public.pl [hunk 1]
```diff
@@ -5,21 +5,32 @@
...
+         ( $searching_user ) = @_;
+
          matrix_create_room( $user,
             preset => "public_chat",
          );
+      })->then( sub {
+         my ( $room_id ) = @_;
+         matrix_join_room( $searching_user, $room_id );
       })->then( sub {
          ( $room_id ) = @_;
```
Comment (richvdh) [inline tests/52user-directory/01public.pl]: this looks bogus. join_room doesn't return a room id. (looks like the function-level $room_id is unused and should be vaped?)

File: tests/52user-directory/01public.pl [hunk 2]
```diff
@@ -44,17 +55,22 @@
       my ( $user ) = @_;
 
       my $room_id;
```
Comment (richvdh) [inline tests/52user-directory/01public.pl]: this is redundant I think?

File: tests/52user-directory/01public.pl [hunk 3]
```diff
@@ -44,17 +55,22 @@
...
 
-      matrix_set_displayname( $user, $displayname )
-      ->then( sub {
          matrix_create_room( $user,
             preset => "private_chat",
          );
       })->then( sub {
-         ( $room_id ) = @_;
-
          matrix_get_user_dir_synced( $user, $displayname );
```
Comment (richvdh) [inline tests/52user-directory/01public.pl]: this user still seems to be looking for themselves?

File: tests/52user-directory/01public.pl [hunk 4]
```diff
@@ -384,12 +400,25 @@
...
       my $room_id;
+      my $searching_user;
 
       my $displayname = generate_random_displayname();
       my $second_displayname = generate_random_displayname();
+      my $searching_displayname = generate_random_displayname();
 
       matrix_set_displayname( $user, $displayname )
       ->then( sub {
+
```
Comment (richvdh) [inline tests/52user-directory/01public.pl]: ugly whitespace makes ugly perl uglier

File: tests/52user-directory/01public.pl [hunk 5]
```diff
@@ -384,12 +400,25 @@
...
+         matrix_create_user_on_server( $user->http,
```
Comment (richvdh) [inline tests/52user-directory/01public.pl]: I'm not quite sure why we are creating a user that is never used here?

File: tests/52user-directory/01public.pl [hunk 6]
```diff
@@ -435,30 +466,43 @@ sub generate_random_displayname
 }
 
 
-# Get the user direectory after a change has been made. This creates a new user
+# Get the user directory after a change has been made. This creates a new user
```
Comment (richvdh) [inline tests/52user-directory/01public.pl]:
```suggestion
# Get the user directory after a change has been made. This creates two new users
```

File: tests/52user-directory/01public.pl [hunk 7]
```diff
@@ -435,30 +466,43 @@ sub generate_random_displayname
...
 # and then polls the user directory until we see it. This is to get around the
```
Comment (richvdh) [inline tests/52user-directory/01public.pl]:
```suggestion
# and then polls the user directory until one sees the other. This is to get around the
```

File: tests/52user-directory/01public.pl (hunk 1 above)
Comment (hawkowl) [reply tests/52user-directory/01public.pl]: done

File: tests/52user-directory/01public.pl (hunk 2 above)
Comment (hawkowl) [reply tests/52user-directory/01public.pl]: fixed

File: tests/52user-directory/01public.pl (hunk 3 above)
Comment (hawkowl) [reply tests/52user-directory/01public.pl]: fixed

File: tests/52user-directory/01public.pl (hunk 5 above)
Comment (hawkowl) [reply tests/52user-directory/01public.pl]: they have to join the room so that the dir is populated

Comment (richvdh) [reply tests/52user-directory/01public.pl]: they don't seem to join a room?

**Pull request merged**